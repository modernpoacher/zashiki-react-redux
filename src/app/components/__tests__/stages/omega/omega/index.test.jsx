import React from 'react'
import PropTypes from 'prop-types'
import snapshotOf, {
  getComponentElement
} from 'react-component-snapshot'

import '@testing-library/jest-dom'

import {
  render
} from '@testing-library/react'

import Omega from '#zashiki-react-redux/app/components/stages/omega/omega'

const MOCK_DESCRIPTION = 'MOCK DESCRIPTION'

const MOCK_DEFINITION = {
  meta: {
    uri: '#/'
  },
  elements: {
    title: 'MOCK TITLE',
    description: 'MOCK DESCRIPTION',
    field: {
      id: 'MOCK ID'
    }
  }
}

const MOCK_RESOURCE = {
  alpha: 'MOCK ALPHA',
  omega: 'MOCK OMEGA'
}

const MOCK_RESPONSE = {
  '#/': 'MOCK VALUE'
}

const MOCK_ERRORS = []

/**
 *  @param {{ to: string | { pathname: string }, children: React.ReactNode | React.ReactNode[] }} props
 *  @returns {React.JSX.Element}
 */
function MockLink ({ to, children }) {
  if (typeof to === 'string') {
    return (
      <a href={to} className='mock-link'>
        {children}
      </a>
    )
  }

  const {
    pathname
  } = to

  return (
    <a href={pathname} className='mock-link'>
      {children}
    </a>
  )
}

MockLink.propTypes = {
  to: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({})
  ]),
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(
      PropTypes.node
    )
  ])
}

jest.mock('react-router', () => {
  return {
    __esModule: true,
    Link: MockLink
  }
})

describe('#zashiki-react-redux/app/components/stages/omega/omega', () => {
  describe('Always', () => {
    it('renders', () => {
      expect(snapshotOf(getComponentElement(render(
        <Omega
          description={MOCK_DESCRIPTION}
          definition={MOCK_DEFINITION}
          resource={MOCK_RESOURCE}
          response={MOCK_RESPONSE}
          errors={MOCK_ERRORS}
          onChange={jest.fn()}
          onSubmit={jest.fn()}
        />
      ))))
        .toMatchSnapshot()
    })
  })
})
