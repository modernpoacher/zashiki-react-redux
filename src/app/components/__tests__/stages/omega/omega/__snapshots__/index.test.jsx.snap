// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#zashiki-react-redux/app/components/stages/omega/omega Always renders 1`] = `
<form
  action="/MOCK ALPHA/MOCK OMEGA"
  method="POST"
  onSubmit={[Function]}
>
  <div
    className="cog text"
  >
    <label
      className="title text"
      htmlFor="MOCK ID"
    >
      <span
        className="text-content"
      >
        MOCK TITLE
      </span>
    </label>
    <span
      className="description text"
    >
      <span
        className="text-content"
      >
        MOCK DESCRIPTION
      </span>
    </span>
    <input
      className="input text"
      disabled={false}
      id="MOCK ID"
      name="MOCK ID"
      onChange={[Function]}
      readOnly={false}
      required={false}
      type="text"
    />
  </div>
  <button
    type="submit"
  >
    Continue
  </button>
</form>
`;
